nomad/job_endpoint_test.go

```diff
@@ -6858,87 +6873,103 @@ func TestJobEndpoint_Dispatch(t *testing.T) {
 
 			var dispatchResp structs.JobDispatchResponse
 			dispatchErr := msgpackrpc.CallWithCodec(codec, "Job.Dispatch", tc.dispatchReq, &dispatchResp)
+			if dispatchErr != nil {
+				must.True(t, tc.err)
+				must.ErrorContains(t, dispatchErr, tc.errStr)
+				return
+			} else if tc.err {
+				t.Logf("Expected error: %v, got: %+v", dispatchErr, dispatchResp)
+			}
 
-			if dispatchErr == nil {
-				if tc.err {
-					t.Fatalf("Expected error: %v", dispatchErr)
+			// Check that we got an eval and job id back
+			switch dispatchResp.EvalID {
+			case "":
+				must.Eq(t, tc.noEval, true)
+				t.Logf("Bad response")
+			default:
+				if tc.noEval {
+					must.Eq(t, tc.noEval, false)
+					t.Logf("Got eval %q", dispatchResp.EvalID)
 				}
+			}
 
-				// Check that we got an eval and job id back
-				switch dispatchResp.EvalID {
-				case "":
-					if !tc.noEval {
-						t.Fatalf("Bad response")
-					}
-				default:
-					if tc.noEval {
-						t.Fatalf("Got eval %q", dispatchResp.EvalID)
-					}
-				}
+			if dispatchResp.DispatchedJobID == "" {
+				t.Fatalf("Bad response")
+			}
 
-				if dispatchResp.DispatchedJobID == "" {
-					t.Fatalf("Bad response")
-				}
+			state := s1.fsm.State()
+			ws := memdb.NewWatchSet()
+			out, err := state.JobByID(ws, tc.parameterizedJob.Namespace, dispatchResp.DispatchedJobID)
+			must.NoError(t, err)
+			if err != nil {
+				t.Logf("err: %v", err)
+			}
 
-				state := s1.fsm.State()
-				ws := memdb.NewWatchSet()
-				out, err := state.JobByID(ws, tc.parameterizedJob.Namespace, dispatchResp.DispatchedJobID)
-				if err != nil {
-					t.Fatalf("err: %v", err)
-				}
-				if out == nil {
-					t.Fatalf("expected job")
-				}
-				if out.CreateIndex != dispatchResp.JobCreateIndex {
-					t.Fatalf("index mis-match")
-				}
-				if out.ParentID != tc.parameterizedJob.ID {
-					t.Fatalf("bad parent ID")
-				}
-				if !out.Dispatched {
-					t.Fatal("expected dispatched job")
-				}
-				if out.IsParameterized() {
-					t.Fatal("dispatched job should not be parameterized")
-				}
-				if out.ParameterizedJob == nil {
-					t.Fatal("parameter job config should exist")
-				}
+			must.NotNil(t, out)
+			if out == nil {
+				t.Logf("expected job, got nil")
+			}
 
-				// Check that the existing job is returned in the case of a supplied idempotency token
-				if tc.idempotencyToken != "" && tc.existingIdempotentJob != nil {
-					if dispatchResp.DispatchedJobID != initialIdempotentDispatchResp.DispatchedJobID {
-						t.Fatal("dispatched job id should match initial dispatch")
-					}
+			must.Eq(t, out.CreateIndex, dispatchResp.JobCreateIndex)
+			if out.CreateIndex != dispatchResp.JobCreateIndex {
+				t.Logf("index mis-match")
+			}
```

Comment:
We've already asserted this above, which will log. If they weren't equal, this block would be unreachable anyways.

```suggestion
```

Similar for other logs in this test.